independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -455,6 +458,22 @@ private void addClassLevelBindings(ClassInfo classInfo, Collection<AnnotationIns
         }
     }
 
+    private void addConstructorLevelBindings(ClassInfo classInfo, Collection<AnnotationInstance> bindings) {
+        MethodInfo constructor = classInfo.method(Methods.INIT);
```

Comment:
This can return any constructor - you first need to look for `@Inject` and if not present, then for no-args one.

Reply:
Done